rql/src/dao.rs

```diff
@@ -383,6 +404,39 @@ impl Dao {
         Ok(records)
     }
 
+    // returns a where clause against all string columns, OR'd,
+    // with the provided query. None is returned if no columns
+    // can be queried against.
+    //
+    // A better query would allow per-column constraints and be
+    // SQL type aware.
+    fn where_clause(&self, schema: &TableSchema, query: &str) -> Option<String> {
+        let constraints: Vec<String> = schema
+            .cols
+            .iter()
+            .filter_map(|c| {
+                let spec = match c {
+                    TableColumn::Spec(spec) => spec,
+                    _ => return None,
+                };
+
+                warn!("constraints type: {}", spec.typ);
+                if spec.typ != "TEXT" {
+                    warn!("fyi, not string: {:?}", spec);
+                    return None;
+                }
+
+                Some(format!(r#"{} LIKE '%{}%'"#, spec.name, query))
+            })
+            .collect();
+
+        if constraints.len() > 0 {
+            Some(constraints.join(" OR "))
```

Comment:
It would :D. I've written a couple of search things like this, and I've previously let it take expressions almost like key-value pairs. 

name=foo|id=bar

But scope bloat, later!